haystack_experimental/components/tools/toolinvoker.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from typing import List
+
+from haystack import component
+
+from haystack_experimental.dataclasses import ChatMessage, ChatRole
+from haystack_experimental.dataclasses.tool import Tool, ToolInvocationError
+
+_FUNCTION_NAME_FAILURE = (
+    "I'm sorry, I tried to run a function that did not exist. Would you like me to correct it and try again?"
+)
+_FUNCTION_RUN_FAILURE = "Seems there was an error while running the function: {error}"
+
+
+@component
+class ToolInvoker:
+    def __init__(self, tools: List[Tool], raise_on_failure: bool = False):
+        self.tools = tools
+        self._tool_names = {tool.name for tool in tools}
+        self.raise_on_failure = raise_on_failure
+
+    @component.output_types(tool_results=List[ChatMessage])
+    def run(self, tool_message: ChatMessage):
```

Comment:
Since the input message can contain one or more tool calls, I think it makes sense to return a list of (tool) messages, each of them wrapping a tool result.